_posts/2023-09-06-#28165.md

```diff
@@ -10,15 +10,80 @@ status: upcoming
+      [responding](https://github.com/bitcoin/bitcoin/blob/083316c4fe20819fbe627c5d21f1a627e10af329/src/net_processing.cpp#L4677-L4694)
+    to `ping` with a `pong` containing the same nonce and
+[disconnecting](https://github.com/bitcoin/bitcoin/blob/083316c4fe20819fbe627c5d21f1a627e10af329/src/net_processing.cpp#L4074-L4077)
+a peer that sends a transaction after we told them not to.
+
+    - The "net" layer "below" net processing abstracts away the details of converting
+      messages to/from bytes that are sent and received on the connection.
+
+- Within the `ProcessMessage` function which responds to pings with pongs, The `ping` is a
+  `CNetMessage` retrieved from `CNode::PollMessage`, and the `pong` is sent by calling
+  `CConnman::PushMessage`.
+
+- [`CConnman::SocketSendData`](https://github.com/bitcoin/bitcoin/blob/master/src/net.cpp#L836) is
+  the function that actually sends data over the connection. It is called by the [socket handler
+thread](https://github.com/bitcoin/bitcoin/blob/083316c4fe20819fbe627c5d21f1a627e10af329/src/net.cpp#L2416)
+and sometimes by [message handler
+thread](https://github.com/bitcoin/bitcoin/blob/083316c4fe20819fbe627c5d21f1a627e10af329/src/net.cpp#L2441)
+from `PushMessage` (an "opportunistic write") if the message queue was empty.
+
+- This PR is a prerequisite of [PR #28196](https://github.com/bitcoin/bitcoin/pull/28196)
+  implementing [BIP324](https://github.com/bitcoin/bips/blob/master/bip-0324.mediawiki), a new
+transport version. While nodes still send the same types of messages and respond to pings with pongs
+in the same way, the way messages are converted to/from bytes is very different.
+
+- This PR creates an abstract `Transport` class, combining the current `TransportDeserializer` and
+  `TransportSerializer` classes into a single interface for converting bytes to/from messages.
+
+- A combined interface means that `V2Transport` can have state shared between the
+serializer and deserializer, such as the `BIP324Cipher` for encrypting/decrypting messages. It may
+be helpful to see how `V2Transport` is implemented in that PR.  Also see a [previous
+approach](https://github.com/bitcoin/bitcoin/pull/24545), which kept the serializer and deserializer
+separate and
+[stored](https://github.com/bitcoin/bitcoin/pull/24545/files#diff-422879cc8bfac56d4380c865f381b58afeb344bc355bbc7f47c581e4491b6b4bR551)
+state in the `CNode` object.
+
 ## Questions
+
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
--->
 
+1. What is the distinction between "net" and "net processing"? What data structures and tasks might
+   we associate with each of them?
+
+1. What does it mean for `CNetMessage` to be "transport protocol agnostic"?
+
+1. When we call `PushMessage` from `SendMessages`, have we sent the bytes corresponding to this
+   message to the peer already (yes/no/maybe)? Why?
```

Comment:
Think it might be worth clarifying exactly what "when" refers to, i.e. is it when we call those functions but have not yet executed (part of) the function, or at the end of the function call?

Reply:
Added